app/src/main/java/com/launcher/silverfish/common/Utils.java

```diff
@@ -26,14 +26,25 @@
 import android.os.AsyncTask;
 import android.util.Log;
 import android.view.Display;
+import android.view.DragEvent;
 import android.widget.ImageView;
 
+import java.lang.ref.WeakReference;
+
 /**
  * Created by Stanislav Pintjuk on 8/3/16.
  * E-mail: stanislav.pintjuk@gmail.com
  */
 public class Utils {
 
+    /**
+     * Is supplied drag event within the 'move to home page' zone?
+     */
+    public static boolean isBeyondRightHandThreshold(Activity a, DragEvent dragEvent) {
+        int xThreshold = Utils.getScreenDimensions(a).x - Constants.SCREEN_CORNER_THRESHOLD;
```

Comment:
Is that SCREEN_CORNER_THRESHOLD value sufficiently high in practice?
Just wondering if there were not too much failed attempts to drag the icon out of the screen when you tried it. :)